apisix/plugins/kafka-logger.lua

```diff
@@ -141,6 +146,29 @@ function _M.log(conf, ctx)
         return
     end
 
+    -- reuse producer via lrucache to avoid unbalanced partitions of messages in kafka
+    local broker_list = core.table.new(0, core.table.nkeys(conf.broker_list))
```

Comment:
The `broker_list` is an array-like table, you should initialize its array part, not the hash table part.